### PR TITLE
fix(build): fix precommit lint hook issue

### LINKED ISSUE #1775: Lint command is not fixing properly the code
## 💥 Regression Report

In this [PR](https://github.com/HospitalRun/hospitalrun-frontend/pull/1736) we discovered that the code wasn't formatted properly. 

## Last working version

Worked up to version: We don't really know

## To Reproduce

* Commit some code that is clearly not properly formatted, ex. `import {cuzz} from 'cuzzo'` instead of the correct`import { cuzz } from 'cuzzo'`.

## Expected behavior
In the above example the code should be formatted and committed correctly. 
```typescript
`import { cuzz } from 'cuzzo'`
```

### PR DESCRIPTION
**Pull request by ocBruno**
fix #1775

**Changes proposed in this pull request:**
- added pre-commit hook and fix lint-staged config
```
"husky": {
    "hooks": {
      "pre-commit": "lint-staged && yarn test:ci",
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
    }
  },
"lint-staged": {
    "*.{js,jsx,ts,tsx}": [
      "eslint --fix"
    ],
    "*.js": "eslint --cache --fix"
  }
```

### CODE REVIEW

**Comment by jackcmeyer** [comment]
Adding the pre-commit hook is a good idea, @ocBruno!

However, I'm not sure that this fixes the initial issue. What was happening is that the build was not failing if there was lint issues because `yarn lint` runs `eslint .... --fix` during the build so any fixable lint issues are not reported on.

There are ways around precommit hooks like using `--no-verify` that would still allow for lint errors to get committed and pushed. 

I think that during the build we need to just run `eslint ...` (without the fix) to ensure builds fail when fixable lint issues are committed and pushed. 

@tehKapa @fox1t what do you think?

**Comment by ocBruno** [comment]
> Adding the pre-commit hook is a good idea, @ocBruno!
> 
> However, I'm not sure that this fixes the initial issue. What was happening is that the build was not failing if there was lint issues because `yarn lint` runs `eslint .... --fix` during the build so any fixable lint issues are not reported on.
> 
> There are ways around precommit hooks like using `--no-verify` that would still allow for lint errors to get committed and pushed.
> 
> I think that during the build we need to just run `eslint ...` (without the fix) to ensure builds fail when fixable lint issues are committed and pushed.
> 
> @tehKapa @fox1t what do you think?



> Adding the pre-commit hook is a good idea, @ocBruno!
> 
> However, I'm not sure that this fixes the initial issue. What was happening is that the build was not failing if there was lint issues because `yarn lint` runs `eslint .... --fix` during the build so any fixable lint issues are not reported on.
> 
> There are ways around precommit hooks like using `--no-verify` that would still allow for lint errors to get committed and pushed.
> 
> I think that during the build we need to just run `eslint ...` (without the fix) to ensure builds fail when fixable lint issues are committed and pushed.
> 
> @tehKapa @fox1t what do you think?

I was removing the --fix option and ran into another issue with eslint and prettier config. 
I was receiving errors on every line saying i needed to remove CR so I did some troubleshooting and realized some misisng configs.
Eslint-config-prettier was present in the packages but not in the last place of the eslint extends array as presented below.
https://prettier.io/docs/en/integrating-with-linters.html
Prettier was missing the rules array as well so I added it. 
I think this resolves some build issues but not sure if the structure is correct.

File: package.json [hunk 1]
```diff
@@ -120,6 +120,7 @@
   },
   "husky": {
     "hooks": {
+      "pre-commit": "lint-staged && yarn test && yarn test:ci",
```
Comment (jackcmeyer) [inline package.json]: I don't think that we need to run `yarn test && yarn test:ci`. `yarn test:ci` should be sufficient

File: .prettierrc [hunk 2]
```diff
@@ -6,5 +6,6 @@
   "trailingComma": "all",
   "bracketSpacing": true,
   "jsxBracketSameLine": false,
-  "arrowParens": "always"
-}
\ No newline at end of file
+  "arrowParens": "always",
+  "endOfLine": "auto"
```
Comment (jackcmeyer) [inline .prettierrc]: I think that we want `"endOfLine"` to be `LF` to be consistent with unix machines

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -6,7 +6,6 @@ import { useTranslation } from 'react-i18next'
 const Navbar = () => {
   const { t } = useTranslation()
   const history = useHistory()
```
Comment (jackcmeyer) [inline src/components/Navbar.tsx]: interesting that the whitespace that got remove din this file...

File: package.json [hunk 4]
```diff
@@ -129,10 +130,9 @@
...
   "lint-staged": {
-    "**/*.{js,jsx,ts,tsx}": [
-      "npm run lint",
-      "npm run test:ci",
-      "git add ."
-    ]
+    "*.{js,jsx,ts,tsx}": [
+      "eslint"
+    ],
+    "*.js": "eslint --cache"
```
Comment (jackcmeyer) [inline package.json]: Is it possible for this config to be equal to the one from the components repo

```json
  "lint-staged": {
    "**/*.{js,jsx,ts,tsx}": [
      "npm run lint",
      "git add ."
    ]
  },
```

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -6,7 +6,6 @@ import { useTranslation } from 'react-i18next'
...
-
```
Comment (jackcmeyer) [inline src/components/Navbar.tsx]: weird that whitespace got removed from this file.

File: package.json (hunk 4 above)
Comment (jackcmeyer) [inline package.json]: Is it possible for this to be the same config as the components repository: 

```json
  "lint-staged": {
    "**/*.{js,jsx,ts,tsx}": [
      "npm run lint",
      "git add ."
    ]
  },
```

File: .prettierrc (hunk 2 above)
Comment (fox1t) [reply .prettierrc]: I agree, we absolutely need consistent endOfLine. You can see this setting also here: https://github.com/HospitalRun/hospitalrun-frontend/blob/master/.editorconfig#L9

File: package.json (hunk 4 above)
Comment (fox1t) [reply package.json]: @tehKapa I want you thoughts on this.

Comment (ocBruno) [reply package.json]: I also think it should be applied to components library, not sure if that period in git add . is redundant and if that cached lint is worth it but I do believe we should add a pre commit test to the components project

**Comment by fox1t** [comment]
> Adding the pre-commit hook is a good idea, @ocBruno!
> 
> However, I'm not sure that this fixes the initial issue. What was happening is that the build was not failing if there was lint issues because `yarn lint` runs `eslint .... --fix` during the build so any fixable lint issues are not reported on.
> 
> There are ways around precommit hooks like using `--no-verify` that would still allow for lint errors to get committed and pushed.
> 
> I think that during the build we need to just run `eslint ...` (without the fix) to ensure builds fail when fixable lint issues are committed and pushed.
> 
> @tehKapa @fox1t what do you think?

@jackcmeyer you are write. Removing --fix ONLY on CI pipeline might be the correct fix.

File: .eslintrc.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ module.exports = {
     'prettier',
     'prettier/@typescript-eslint',
     'plugin:prettier/recommended',
+    'eslint-config-prettier'
```
Comment (tehKapa) [inline .eslintrc.js]: We run prettier as an eslint plugin, so this harmful

File: .eslintrc.js [hunk 7]
```diff
@@ -31,6 +32,7 @@ module.exports = {
   },
   plugins: ['react', '@typescript-eslint', 'prettier', 'jest'],
   rules: {
+    "prettier/prettier": "error",
```
Comment (tehKapa) [inline .eslintrc.js]: `'plugin:prettier/recommended'` already includes this rule. Refence: https://github.com/prettier/eslint-plugin-prettier#recommended-configuration

File: .prettierrc [hunk 8]
```diff
@@ -6,5 +6,6 @@
   "trailingComma": "all",
   "bracketSpacing": true,
   "jsxBracketSameLine": false,
-  "arrowParens": "always"
-}
\ No newline at end of file
+  "arrowParens": "always",
+  "endOfLine": "lf"
```
Comment (tehKapa) [inline .prettierrc]: I force this to lf

File: package.json [hunk 9]
```diff
@@ -129,11 +130,10 @@
       "path": "./node_modules/cz-conventional-changelog"
     }
   },
-  "lint-staged": {
+    "lint-staged": {
     "**/*.{js,jsx,ts,tsx}": [
       "npm run lint",
-      "npm run test:ci",
```
Comment (tehKapa) [inline package.json]: it's important to test commit before add files to stage

File: package.json [hunk 10]
```diff
@@ -121,6 +121,7 @@
   },
   "husky": {
     "hooks": {
+      "pre-commit": "lint-staged && yarn test:ci",
```
Comment (tehKapa) [inline package.json]: see the next comment (remove `&& yarn test:ci`)

Comment (ocBruno) [reply package.json]: Applied the changs but am receiving this error on commit.
```

× npm found some errors. Please fix them and try committing again.

> @hospitalrun/frontend@0.0.0-development test:ci C:\Users\55419\Code\hospitalrun-frontend
> cross-env CI=true react-scripts test "C:/Users/55419/Code/hospitalrun-frontend/.eslintrc.js"

No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In C:\Users\55419\Code\hospitalrun-frontend
82 files checked.
testMatch: C:/Users/55419/Code/hospitalrun-frontend/src/**/__tests__/**/*.{js,jsx,ts,tsx}, C:/Users/55419/Code/hospitalrun-frontend/src/**/*.{spec,test}.{js,jsx,ts,tsx} - 30 matches
testPathIgnorePatterns: \
ode_modules\\ - 82 matches
testRegex:  - 0 matches
Pattern: C:\\Users\\55419\\Code\\hospitalrun-frontend\\.eslintrc.js - 0 matches
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @hospitalrun/frontend@0.0.0-development test:ci: `cross-env CI=true react-scripts test "C:/Users/55419/Code/hospitalrun-frontend/.eslintrc.js"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @hospitalrun/frontend@0.0.0-development test:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\55419\AppData\Roaming
pm-cache\_logs\2020-02-07T13_30_53_752Z-debug.log
husky > pre-commit hook failed (add --no-verify to bypass
```

Comment (tehKapa) [reply package.json]: I [pushed](https://github.com/HospitalRun/hospitalrun-frontend/pull/1780/commits/639f122b11e6eee2953d0ad40db6edbd22b82f61) changes myself and didn't have any trouble.

**Pull request merged**